internal/buildengine/engine.go

```diff
@@ -755,7 +755,7 @@ func (e *Engine) getDependentModuleNames(moduleName string) []string {
 }
 
 // BuildAndDeploy attempts to build and deploy all local modules.
-func (e *Engine) BuildAndDeploy(ctx context.Context, replicas int32, waitForDeployOnline bool, moduleNames ...string) error {
+func (e *Engine) BuildAndDeploy(ctx context.Context, replicas int32, waitForDeployOnline bool, timeout optional.Option[time.Duration], moduleNames ...string) error {
```

Comment:
It's _much_ more idiomatic to use the context for timeouts.

Reply:
Fixed